### ISSUE TITLE
Stickers fail to send / Stickers are corrupted

### ISSUE THREAD

**Issue by wnhre2ur8cxx8** [signalapp/Signal-Android#9931]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Some stickers are not sent or received. It feels random but is very very common for me. It is happenening since the last 3-4 version upgrades.

### Steps to reproduce
- Send a sticker (to myself or somebody else)
- The sticker does not show up, instead it is just blank with the exclamation mark in the bottom
- "Send failed" in the notification bar
- Someti9mes it works though, I also have this problem when I get a sticker sent

**Actual result:** "Send failed" No sticker seen
**Expected result:** A sent or received sticker

### Device info

**Device:** Xiaomi Mi A1
**Android version:** 9
**Signal version:** 4.68.8

### Link to debug log
https://debuglogs.org/07b50a66b0a65d1a1e933b114381c7f8643355418ffe1c2f7370aa5da481d38e

**Comment by greyson-signal** [comment]
Looks like we may be having trouble reading the sticker:

```java
2020-08-17 19:57:47.163 CEST W MediaConstraints: Failed to determine if media's constraints are satisfied.
2020-08-17 19:57:47.163 CEST W MediaConstraints: java.io.IOException: org.thoughtcrime.securesms.util.BitmapDecodingException: Failed to decode image dimensions: -1, -1
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.mms.MediaConstraints.isWithinBounds(MediaConstraints.java:69)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.mms.MediaConstraints.isSatisfied(MediaConstraints.java:52)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.jobs.AttachmentCompressionJob.scaleAndStripExif(AttachmentCompressionJob.java:152)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.jobs.AttachmentCompressionJob.onRun(AttachmentCompressionJob.java:126)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.jobs.BaseJob.run(BaseJob.java:25)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.jobmanager.JobRunner.run(JobRunner.java:85)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.jobmanager.JobRunner.run(JobRunner.java:48)
2020-08-17 19:57:47.163 CEST W MediaConstraints: Caused by: org.thoughtcrime.securesms.util.BitmapDecodingException: Failed to decode image dimensions: -1, -1
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.util.BitmapUtil.getImageDimensions(BitmapUtil.java:225)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.util.BitmapUtil.getDimensions(BitmapUtil.java:252)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	at org.thoughtcrime.securesms.mms.MediaConstraints.isWithinBounds(MediaConstraints.java:65)
2020-08-17 19:57:47.163 CEST W MediaConstraints: 	... 6 more
2020-08-17 19:57:47.183 CEST W BaseJob: [JOB::0507c2c7-9286-4db9-995c-b888b875b925][AttachmentCompressionJob] Encountered a failing exception. (Time Since Submission: 242 ms, Lifespan: 86400000 ms, Run Attempt: 1/Unlimited)
2020-08-17 19:57:47.183 CEST W BaseJob: org.thoughtcrime.securesms.transport.UndeliverableMessageException: java.io.IOException: org.thoughtcrime.securesms.util.BitmapDecodingException: java.util.concurrent.ExecutionException: com.bumptech.glide.load.engine.GlideException: Failed to load resource
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.jobs.AttachmentCompressionJob.scaleAndStripExif(AttachmentCompressionJob.java:166)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.jobs.AttachmentCompressionJob.onRun(AttachmentCompressionJob.java:126)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.jobs.BaseJob.run(BaseJob.java:25)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.jobmanager.JobRunner.run(JobRunner.java:85)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.jobmanager.JobRunner.run(JobRunner.java:48)
2020-08-17 19:57:47.183 CEST W BaseJob: Caused by: java.io.IOException: org.thoughtcrime.securesms.util.BitmapDecodingException: java.util.concurrent.ExecutionException: com.bumptech.glide.load.engine.GlideException: Failed to load resource
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.jobs.AttachmentCompressionJob.getResizedMedia(AttachmentCompressionJob.java:260)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.jobs.AttachmentCompressionJob.scaleAndStripExif(AttachmentCompressionJob.java:159)
2020-08-17 19:57:47.183 CEST W BaseJob: 	... 4 more
2020-08-17 19:57:47.183 CEST W BaseJob: Caused by: org.thoughtcrime.securesms.util.BitmapDecodingException: java.util.concurrent.ExecutionException: com.bumptech.glide.load.engine.GlideException: Failed to load resource
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.util.BitmapUtil.createScaledBytes(BitmapUtil.java:155)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.util.BitmapUtil.createScaledBytes(BitmapUtil.java:79)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.util.BitmapUtil.createScaledBytes(BitmapUtil.java:67)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.util.BitmapUtil.createScaledBytes(BitmapUtil.java:53)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.jobs.AttachmentCompressionJob.getResizedMedia(AttachmentCompressionJob.java:251)
2020-08-17 19:57:47.183 CEST W BaseJob: 	... 5 more
2020-08-17 19:57:47.183 CEST W BaseJob: Caused by: java.util.concurrent.ExecutionException: com.bumptech.glide.load.engine.GlideException: Failed to load resource
2020-08-17 19:57:47.183 CEST W BaseJob: 	at com.bumptech.glide.request.RequestFutureTarget.doGet(RequestFutureTarget.java:217)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at com.bumptech.glide.request.RequestFutureTarget.get(RequestFutureTarget.java:130)
2020-08-17 19:57:47.183 CEST W BaseJob: 	at org.thoughtcrime.securesms.util.BitmapUtil.createScaledBytes(BitmapUtil.java:105)
2020-08-17 19:57:47.183 CEST W BaseJob: 	... 9 more
2020-08-17 19:57:47.183 CEST W BaseJob: Caused by: com.bumptech.glide.load.engine.GlideException: Failed to load resource
```

Are you able to share the sticker pack(s) and sticker(s) you're having trouble with?

**Comment by wnhre2ur8cxx8** [comment]
Hey sorry for the delay. Holiday :)

The Problem still exists, sometimes they are now sent, but are displayed corrupted on my end.

The problem persists between sticker packs. Its even the official ones the refuse to send. For example the official dog one. It seems to be a problem of single stickers in the pack. For example the "dog partying" sticker (the 4th one) refuses to send every time. Or the 2nd sticker "laughing dog". The 1st one seems to work fine. In general its about a 50% chance for every sticker I have. I created a small video and uploaded it here so you can see what I mean, I also have the debug log shortly after the recording.

[recording_20200921_142416.zip](https://github.com/signalapp/Signal-Android/files/5255324/recording_20200921_142416.zip)

https://debuglogs.org/8249b8fb03d187cb3f803baff86c85192bb1f08d6b6275959fcd156829b6abe8

**Comment by wnhre2ur8cxx8** [comment]
For the corrupted images, it only appeared for me on these sticker packs:

Funny Animals

https://signal.art/addstickers/#pack_id=fe486a1edeb265cde15accf9a090095e&pack_key=7aba7caea1d316b2c8279f02ebf4028b0896dce4813cd50e5d372e7be9658032
1m

Meme Pack

https://signal.art/addstickers/#pack_id=2f977990cc97776e3507f1f8c8ffa1b9&pack_key=9ef33e78c237f9da414e8758af1fba810878ae34c03afea43b2035bcb2d5321a

Broken Cats

https://signal.art/addstickers/#pack_id=cf238b5b86d5e2c3e8f0a39b1731e551&pack_key=0f1f37b2da8c083e63c71573273b18525cdf833eea76075a776867c66612669f

**Comment by wnhre2ur8cxx8** [comment]
@greyson-signal So the issue persists, but I was able to find out that it seems to only happen to "old" sticker packs. At least the new ones I added just recently like: https://signalstickers.com/pack/c4d174fe9e61444fcc9a354c0f33a61c or https://signalstickers.com/pack/ea128ace5dace7b662144d7c9054124c dont have the issue at all.

Just to be clear I tried (right now again) to remove and readd the stickerpacks with the problems. They still have it. But that seems to switch around the broken stickers for some reason. Anyway, have another debug log good sir: https://debuglogs.org/b8efaf59f4e9ff9a3782a78ca3307355e2ed33e65d5479d7c55fd032e43d2e17

**Comment by wnhre2ur8cxx8** [comment]
Lots of updates, still my issue persists.. I even switched phone in between. I am on Android 10 now. Signal backup restored. So I guess my database somehow is broken?

**Comment by wnhre2ur8cxx8** [comment]
Unfortunately the issue still persists. I can do another bug report with an up to date signal. But only if there is a need for it.

**Comment by alex-signal** [comment]
If possible, try removing one of the offending sticker packs and redownloading it. It's likely that the downloaded pack itself is corrupt.

**Comment by wnhre2ur8cxx8** [comment]
Ok, I did that in the past, but will try it again and report. Thanks for the fast response :)

**Comment by wnhre2ur8cxx8** [comment]
Allright, apparently most of the stickers now work fine. I found one thats still corrupted though. Its from your official Stickers, "Swoon / Hands" and its the crossed fingers one. I removed it and installed it again but the outcome didnt change. What changed though is that the stickers are now sent properly (both checkmarks). And on the receiving side they are displayed correctly.

Do you need screenshots or a log or anything?

**Comment by wnhre2ur8cxx8** [comment]
Ah sorry, the removing and installing thing  didnt change that they are sent properly. It was sent normally before doing that aswell.

**Comment by cody-signal** [comment]
So is this still an issue or is it working, sorry, I'm having trouble tracking what happened. If it is still happening, an updated debuglog is always helpful.

**Comment by wnhre2ur8cxx8** [comment]
Sorry it took me so long, I guess also the bug is not super important, but it has been existing since I first posted the issue so I guess there is still something broken. I even switched phones, I switched phone number, reinstalled signal, etc.

I finally created a new debug log: https://debuglogs.org/android/5.35.3/2860e104c69a010195dcae7999e5d90cb6e9c52495f20398b0982e8556f592c0

What I did this time is, i sent some stickers that I knew were working, and the I sent a sticker that was corrupted. All were in Swoons hands sticker pack, so the official one. Then I removed the sticker pack and added it again. then sent the broken stickers again.

On my desktop devices there is no corruption by the way. the stickers send just fine.
